### PR TITLE
Eeprom v7 flash update in calibration

### PR DESCRIPTION
**Pull request by saching13**

### CODE REVIEW

File: calibrate.py [hunk 1]
```diff
@@ -540,8 +540,9 @@ def calibrate(self):
                 left = dai.CameraBoardSocket.RIGHT
                 right = dai.CameraBoardSocket.LEFT
 
-            calibration_handler = dai.CalibrationHandler()
-            calibration_handler.setBoardInfo(self.board_config['board_config']['name'], self.board_config['board_config']['revision'])
+            calibration_handler = self.device.readCalibration2()
```
Comment (szabi-luxonis) [inline calibrate.py:543]: Bump depthai to 2.15.5.0 in requirements

**Review by szabi-luxonis — CHANGES_REQUESTED** [review]

File: calibrate.py (hunk 1 above)
Comment (themarpe) [inline calibrate.py:543]: readCalibration2 throws if EEPROM is empty. Make sure to catch / notify / handle that case.
Also, make sure to flash both USER & FACTORY sections and compare those to be equal

**Review by themarpe — CHANGES_REQUESTED** [review]

**Comment by Erol444** [comment]
@saching13 I also created a branch for that; https://github.com/luxonis/depthai/pull/708 Maybe we could use that, as it will flash name/revision only if it's empty? It was also [tested by one user](https://discord.com/channels/790680891252932659/924798475030392922/980828130858127421).

**Comment by saching13** [comment]
> @saching13 I also created a branch for that; #708 Maybe we could use that, as it will flash name/revision only if it's empty? It was also [tested by one user](https://discord.com/channels/790680891252932659/924798475030392922/980828130858127421).

Sounds good. @Erol444  do you want to make that PR or should I import them here and wrap it up ?

**Review by themarpe — COMMENTED** [review]
Is the "factory flashing" limited to factory devices? Or do you intend to user expose it? @saching13

**Review by themarpe — CHANGES_REQUESTED** [review]
`readCalibration2` could be used instead of having to cross check "default" calibration returned by `readCalibration` against it being empty.

Otherwise lgtm

**Review by themarpe — APPROVED** [review]

**Pull request merged**